barcode_action/static/src/js/action_barcode_form.js

```diff
@@ -0,0 +1,28 @@
+odoo.define('barcode_action.form', function (require) {
+"use strict";
+
+var FormController = require('web.FormController');
+
+FormController.include({
+    _barcodeHandleAction: function (barcode, activeBarcode) {
+        if (this.mode === 'readonly') {
+            this.do_warn(_t('Error : Document not editable'),
```

Comment:
Is this check necessary?

Reply:
Mmmmh, good point, I followed the same logic used on [odoo](https://github.com/odoo/odoo/blob/11.0/addons/barcodes/static/src/js/barcode_form_view.js#L62), but I could remove it if you feel that it is unnecessary.

Reply:
Removed